From c61a2ce253949266a948e094c31f3a65068f0634 Mon Sep 17 00:00:00 2001
From: Yidi Lin <yidilin@google.com>
Date: Fri, 13 Mar 2026 21:05:18 +0800
Subject: [PATCH 50/63] ANDROID: ABI: Add region_intersects to the symbol table

Add region_intersects to the symbol table in order to check no-map
memory region in coreboot_table module.

Bug: 479732141
Bug: 498573387
Test: Build

Change-Id: Ie58dfeb5b16ea0aaef1b7323b766ababbb81b65f
Signed-off-by: Yidi Lin <yidilin@google.com>
(cherry picked from commit 17626f3cc72278b82998ad8d51c6d01ea62d61ef)
---
 gki/aarch64/abi.stg         | 18 ++++++++++++++++++
 gki/aarch64/symbols/desktop |  1 +
 2 files changed, 19 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 73e94e6d0298..4a3e4da2a160 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -430389,6 +430389,14 @@ function {
   parameter_id: 0x1344d43c
   parameter_id: 0xf1a6dfed
 }
+function {
+  id: 0xbb057b65
+  return_type_id: 0x6720d32f
+  parameter_id: 0x9565759f
+  parameter_id: 0xf435685e
+  parameter_id: 0x33756485
+  parameter_id: 0x33756485
+}
 function {
   id: 0xbb40a305
   return_type_id: 0x6720d32f
@@ -502322,6 +502330,15 @@ elf_symbol {
   type_id: 0x9eed9829
   full_name: "regcache_sync_region"
 }
+elf_symbol {
+  id: 0x82f8f67b
+  name: "region_intersects"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x8f7f8afd
+  type_id: 0xbb057b65
+  full_name: "region_intersects"
+}
 elf_symbol {
   id: 0xceab3170
   name: "register_btf_kfunc_id_set"
@@ -533790,6 +533807,7 @@ interface {
   symbol_id: 0x60b782b4
   symbol_id: 0x38aaa220
   symbol_id: 0xb34d0634
+  symbol_id: 0x82f8f67b
   symbol_id: 0xceab3170
   symbol_id: 0xeab63d27
   symbol_id: 0x1bc307a0
diff --git a/gki/aarch64/symbols/desktop b/gki/aarch64/symbols/desktop
index 5ce3c75ea8ae..3ce862edf434 100644
--- a/gki/aarch64/symbols/desktop
+++ b/gki/aarch64/symbols/desktop
@@ -1637,6 +1637,7 @@
   regcache_cache_only
   regcache_mark_dirty
   regcache_sync
+  region_intersects
   __register_chrdev
   register_chrdev_region
   register_die_notifier
-- 
2.50.1

